Services/UI/OJS.Services.Ui.Data/Implementations/SubmissionsDataService.cs

```diff
@@ -26,107 +27,107 @@ public Task<IEnumerable<TServiceModel>> GetLatestSubmissions<TServiceModel>(int
             .ToEnumerableAsync();
 
     public Submission? GetBestForParticipantByProblem(int participantId, int problemId) =>
-            this.GetAllByProblemAndParticipant(problemId, participantId)
-                .Where(s => s.Processed)
-                .OrderByDescending(s => s.Points)
-                .ThenByDescending(s => s.Id)
-                .FirstOrDefault();
+        this.GetAllByProblemAndParticipant(problemId, participantId)
+            .Where(s => s.Processed)
+            .OrderByDescending(s => s.Points)
+            .ThenByDescending(s => s.Id)
+            .FirstOrDefault();
+
+    public IQueryable<Submission> GetByIdQuery(int id) =>
+        this.DbSet
+            .Where(s => s.Id == id);
+
+    public IQueryable<Submission> GetAllByProblem(int problemId)
```

Comment:
`Task<IEnumerable<T>>`.
The rest as well

Reply:
As discussed I am leaving the methods that were already created as they were. The method I created GetAllByProblemAndUser returns Task<IEnumerable<TServiceModel>>.